xgcm/axis.py

```diff
@@ -0,0 +1,208 @@
+        name : str
+            Name of this Axis.
+        coords : dict
+            Mapping of axis positions to coordinate names
+            (e.g. `{'center': 'XC', 'left: 'XG'}`)
+        default_shifts : dict, optional
+            Default mapping from and to grid positions
+            (e.g. `{'center': 'left'}`). Will be inferred if not specified.
+        boundary : str or dict
+            boundary can either be one of {None, 'fill', 'extend', 'extrapolate', 'periodic'}
+            * None:  Do not apply any boundary conditions. Raise an error if
+              boundary conditions are required for the operation.
+            * 'fill':  Set values outside the array boundary to fill_value
+              (i.e. a Dirichlet boundary condition.)
+            * 'extend': Set values outside the array to the nearest array
+              value. (i.e. a limited form of Neumann boundary condition where
+              the difference at the boundary will be zero.)
+            * 'periodic' : Wrap arrays around. Equivalent to setting `periodic=True`
+            This sets the default value. It can be overriden by specifying the
+            boundary kwarg when calling specific methods.
+        fill_value : float, optional
+            The value to use in the boundary condition when boundary='fill'.
+        """
+
+        if not isinstance(name, str):
+            raise TypeError(
+                f"name argument must be of type str, but is of type {type(name)}"
+            )
+
+        self._name = name
+
+        if not isinstance(ds, xr.Dataset):
+            raise TypeError(
+                f"ds argument must be of type xarray.Dataset, but is of type {type(ds)}"
+            )
+
+        # check all inputted values are valid here
+        for pos, dim in coords.items():
+            if pos not in VALID_POSITION_NAMES.split("|"):
+                raise ValueError(
+                    f"Axis position must be one of {VALID_POSITION_NAMES.split('|')}, but got {pos}"
+                )
+            if dim not in ds.dims:
+                raise ValueError(
+                    f"Could not find dimension `{dim}` (for the `{pos}` position on axis `{name}`) in input dataset."
+                )
+        self._coords = coords
+
+        # set default position shifts
+
+        if default_shifts is None:
+            default_shifts = {}
+
+        self._default_shifts = {}
+        for pos in self.coords:
+            # use user-specified value if present
+            if pos in default_shifts:
+                self._default_shifts[pos] = default_shifts[pos]
+            else:
+                for possible_shift in FALLBACK_SHIFTS[pos]:
+                    if possible_shift in self.coords:
+                        self._default_shifts[pos] = possible_shift
+                        break
+
+            if pos in self._default_shifts and self._default_shifts[pos] == pos:
+                # TODO stricter checking? e.g. non-adjacent positions?
```

Comment:
Not sure I still understand this comment.

Reply:
Nor me.

Reply:
Then lets remove it I guess?